src/groth16/aggregate/proof.rs

```diff
@@ -1,76 +1,73 @@
-use serde::{Deserialize, Serialize};
-
 use crate::bls::Engine;
+use crate::groth16::aggregate::commit;
 
-#[derive(Serialize, Deserialize)]
```

Comment:
why you remove serialization and deserialization? we still need this

Reply:
Yes... so when I started I made a clean slate and I didn't know how to figure out all these derive required for serde:
* I usually only use `derive(Serialize,Deserialize)` but then the compiler wasn't happy showing me things related to lifetimes: 
```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'de`
due to conflicting requirements
  --> src/groth16/aggregate/proof.rs:29:5
   |
29 |     /// ((T_L, U_L),(T_R,U_R)) values accross all steps
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 ``` (plenty)
I would appreciate any help there as I'm not yet used to the dark magic of serde features with lifetimes ... ><

Reply:
Im looking at it now - seems like I need to specify to serde how to serialize all these fields as he can't figure out the right type to put the bounds `: Deserialize` on.

Reply:
Ok I'm not sure I understand in depth why it's working but I made it working as before now.